### PR TITLE
feat: edit profile's education part to fit the mockup

### PR DESCRIPTION
**Pull request by xiangxinyue**
**Changes**
- Changed the education_item.dart to fit the mockup.
- Changed the images in the codebase to fit the mockup.

**UI**

edit_education

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -566,4 +566,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: c34e2287a9ccaa606aeceab922830efb9a6ff69a
 
-COCOAPODS: 1.9.3
+COCOAPODS: 1.9.1
```
Comment (eivenlour) [inline ios/Podfile.lock:569]: Keep this as `1.9.3` since it's the latest version

File: lib/navigation/app_controller.dart [hunk 2]
```diff
@@ -47,7 +46,8 @@ class _AppControllerState extends State<AppController> {
     uid = userIdSelector(store);
     _screens = [UserProfile(profileId: uid), Pairings(), Chat()];
     return Scaffold(
-        body: _screens[pageIndex],
+        body: 
+        _screens[pageIndex],
```
Comment (eivenlour) [inline lib/navigation/app_controller.dart:50]: Make this one-line

File: lib/profile/save_profile/sections/education/education_item.dart [hunk 3]
```diff
@@ -66,16 +71,32 @@ class _SaveEducationItemState extends State<SaveEducationItem> {
...
-          onSaveActionTap: handleSaveEducation,
-        ),
+        appBar: AppBar(
+            elevation: 0.0,
+            leading: IconButton(
+            iconSize: screenHeight * 0.02,
+            onPressed: () {
+              handleClose();
+              Navigator.pop(context);
+              },
```
Comment (eivenlour) [inline lib/profile/save_profile/sections/education/education_item.dart:92]: Take `Navigator.pop(context)` out here and add it in the `handleClose()` function instead

File: lib/profile/save_profile/sections/education/education_item.dart [hunk 4]
```diff
@@ -338,19 +309,119 @@ class _SaveEducationItemState extends State<SaveEducationItem> {
...
+            name: value,
+            endYear: school.endYear
+          )
+        )
+      ),
+        decoration: fieldDecoration(
+          context,
+          isFocused: isSchoolNameFocused,
+          isInvalid: isSchoolNameInvalid,
+          hintText: "Enter your institution",
```
Comment (eivenlour) [inline lib/profile/save_profile/sections/education/education_item.dart:362]: Add this to the `profile_constants` file

File: lib/profile/save_profile/sections/education/education_item.dart [hunk 5]
```diff
@@ -338,19 +309,119 @@ class _SaveEducationItemState extends State<SaveEducationItem> {
...
+        maxLines: 1,
+        style: fieldTextStyle(color: Theme.of(context).primaryColor),
+        onChanged: (value) => store.dispatch(
+        UpdateSaveEducationState(
+          school.copyWith(
+            name: value,
+            endYear: school.endYear
+          )
+        )
+      ),
```
Comment (eivenlour) [inline lib/profile/save_profile/sections/education/education_item.dart:357]: Fix indentation here

File: lib/profile/save_profile/sections/education/education_item.dart [hunk 6]
```diff
@@ -338,19 +309,119 @@ class _SaveEducationItemState extends State<SaveEducationItem> {
...
+        maxLines: 1,
+        style: fieldTextStyle(color: Theme.of(context).primaryColor),
+        onChanged: (value) => store.dispatch(
+        UpdateSaveEducationState(
+          school.copyWith(
+            program: value,
+            endYear: school.endYear
+          )
+        )
+      ),
```
Comment (eivenlour) [inline lib/profile/save_profile/sections/education/education_item.dart:391]: Fix indentation here

File: lib/profile/save_profile/sections/education/education_item.dart [hunk 7]
```diff
@@ -338,19 +309,119 @@ class _SaveEducationItemState extends State<SaveEducationItem> {
...
+        decoration: fieldDecoration(
+          context,
+          isFocused: isDegreeFocused,
+          isInvalid: isDegreeInvalid,
+          hintText: "Enter your credential",
```
Comment (eivenlour) [inline lib/profile/save_profile/sections/education/education_item.dart:396]: Add this to the `profile_constants` file

File: lib/profile/save_profile/sections/education/education_item.dart [hunk 8]
```diff
@@ -338,19 +309,119 @@ class _SaveEducationItemState extends State<SaveEducationItem> {
...
+      padding: EdgeInsets.only(top:100.0, bottom: 20.0, 
+      left: 60.0, right: 60.0),
+      child: PrimaryButton(
+        isLight: true,
+        text: widget.editMode ? 'Edit' : 'Add',
+        onPressed: () {
+          if (_saveEducationItemKey.currentState.saveAndValidate()) {
+            handleSaveEducation();
+          } else {
+            print("Validation failed.");
```
Comment (eivenlour) [inline lib/profile/save_profile/sections/education/education_item.dart:419]: Merge all these line into the `handleSaveEducation` function

File: lib/profile/save_profile/sections/education/education_item.dart [hunk 9]
```diff
@@ -36,8 +36,13 @@ class _SaveEducationItemState extends State<SaveEducationItem> {
   final FocusNode degreeFocus = FocusNode();
   
   bool isStartYearFocused, isEndYearFocused, isActive;
+
```
Comment (eivenlour) [inline lib/profile/save_profile/sections/education/education_item.dart:39]: Delete this new line!

File: lib/profile/save_profile/sections/education/education_item.dart [hunk 10]
```diff
@@ -8,15 +8,15 @@ import 'package:random_string/random_string.dart';
...
+import '../../../../icons/aspire_icons.dart';
 import '../../../../models/models.dart';
 import '../../../../selectors/selectors.dart';
-import '../../../common/app_bar.dart';
 import '../../../common/date_picker.dart';
-import '../../../common/delete_button.dart';
 import '../../../common/input_field.dart';
 import '../../../common/picker_field.dart';
-import '../../../common/styles.dart';
+
```
Comment (eivenlour) [inline lib/profile/save_profile/sections/education/education_item.dart:19]: Delete this new line!

File: lib/profile/save_profile/sections/education/education_item.dart (hunk 3 above)
Comment (xiangxinyue) [reply lib/profile/save_profile/sections/education/education_item.dart:92]: Got ya

File: lib/profile/save_profile/sections/education/education_item.dart (hunk 8 above)
Comment (xiangxinyue) [reply lib/profile/save_profile/sections/education/education_item.dart:419]: I don't understand this comment.

Comment (HelenGezahegn) [reply lib/profile/save_profile/sections/education/education_item.dart:419]: Move the conditional to the `handleSaveEducation()`! The `onPressed` is too long and we can move the `if` and `else` in `handleSaveEducation`.

File: lib/navigation/app_controller.dart [hunk 11]
```diff
@@ -13,10 +13,10 @@ import '../selectors/selectors.dart';
 
 class AppController extends StatefulWidget {
   final int tabIndex;
-
   AppController({Key key, this.tabIndex}) : super(key: key);
 
   @override
+
```
Comment (HelenGezahegn) [inline lib/navigation/app_controller.dart:19]: Delete this line.

File: lib/navigation/app_controller.dart [hunk 12]
```diff
@@ -58,10 +58,11 @@ class _AppControllerState extends State<AppController> {
           strokeColor: navItemColor, 
           unSelectedColor: Colors.grey,
           backgroundColor: Colors.white,
-        )
+        ),
       );
```
Comment (HelenGezahegn) [inline lib/navigation/app_controller.dart:62]: Fix indentation here.

File: lib/profile/save_profile/sections/education/education_item.dart (hunk 3 above)
Comment (HelenGezahegn) [reply lib/profile/save_profile/sections/education/education_item.dart:92]: Also, fix indentation here with the `onPressed`!

File: lib/profile/save_profile/sections/education/education_item.dart [hunk 13]
```diff
@@ -338,19 +309,119 @@ class _SaveEducationItemState extends State<SaveEducationItem> {
...
+        isLight: true,
+        text: widget.editMode ? 'Edit' : 'Add',
+        onPressed: () {
+          if (_saveEducationItemKey.currentState.saveAndValidate()) {
+            handleSaveEducation();
+          } else {
+            print("Validation failed.");
+          }
+          
+        },
```
Comment (HelenGezahegn) [inline lib/profile/save_profile/sections/education/education_item.dart:422]: Delete this comma here.

File: lib/profile/save_profile/sections/education/education_item.dart [hunk 14]
```diff
@@ -338,19 +309,119 @@ class _SaveEducationItemState extends State<SaveEducationItem> {
...
+      child: PrimaryButton(
+        isLight: true,
+        text: widget.editMode ? 'Edit' : 'Add',
+        onPressed: () {
+          if (_saveEducationItemKey.currentState.saveAndValidate()) {
+            handleSaveEducation();
+          } else {
+            print("Validation failed.");
+          }
+
```
Comment (HelenGezahegn) [inline lib/profile/save_profile/sections/education/education_item.dart:421]: Delete this extra space.

File: lib/profile/save_profile/sections/education/education_item.dart (hunk 8 above)
Comment (eivenlour) [reply lib/profile/save_profile/sections/education/education_item.dart:419]: Your `handleSaveEducation()` function should look like:
```
void handleSaveEducation() {
    if (_saveEducationItemKey.currentState.saveAndValidate()) {
        store.dispatch(SaveSchool(payload: widget.editMode));
        setState(() { isActive = false; });
        unfocusFields();
        Navigator.pop(context);
   } else {
       print("Validation failed.");
  }
```

Then you could just do this instead under the Add/Edit button widget:
```
onPressed: handleSaveEducation
```

**Comment by HelenGezahegn** [comment]
Also a side tip, you can make the image in your description smaller by using the ``

File: lib/profile/save_profile/sections/education/education_item.dart (hunk 8 above)
Comment (xiangxinyue) [reply lib/profile/save_profile/sections/education/education_item.dart:419]: That's make sense!

File: ios/Podfile.lock (hunk 1 above)
Comment (xiangxinyue) [reply ios/Podfile.lock]: How about 1.10.0

File: lib/constants/profile_constants.dart [hunk 15]
```diff
@@ -68,4 +68,9 @@ const String invalidPhoneNumberErrorMessage = 'Enter a valid phone number.';
 
 // ISO codes
 const String isoCodeUS = 'US';
-const String isoCodeCA = 'CA';
\ No newline at end of file
+const String isoCodeCA = 'CA';
+
+// Education Item 
+
```
Comment (eivenlour) [inline lib/constants/profile_constants.dart:74]: Delete this line please!

File: lib/constants/profile_constants.dart [hunk 16]
```diff
@@ -68,4 +68,9 @@ const String invalidPhoneNumberErrorMessage = 'Enter a valid phone number.';
...
+const String buildSchoolNameHint = 'Enter your institution';
+const String buildDegreeFieldHint = 'Enter your credential';
```
Comment (eivenlour) [inline lib/constants/profile_constants.dart:76]: You can just call these strings `schoolNameHint` and `degreeHint`

File: lib/navigation/app_controller.dart (hunk 11 above)
Comment (eivenlour) [inline lib/navigation/app_controller.dart:19]: Echoing @HelenGezahegn's past comment to delete this line

File: lib/navigation/app_controller.dart [hunk 17]
```diff
@@ -58,10 +57,11 @@ class _AppControllerState extends State<AppController> {
           strokeColor: navItemColor, 
           unSelectedColor: Colors.grey,
           backgroundColor: Colors.white,
-        )
-      ); 
+        ),
```
Comment (eivenlour) [inline lib/navigation/app_controller.dart:60]: Get rid of this comma please!

File: lib/profile/save_profile/sections/education/education_item.dart [hunk 18]
```diff
@@ -37,8 +36,13 @@ class _SaveEducationItemState extends State<SaveEducationItem> {
   final FocusNode degreeFocus = FocusNode();
   
   bool isStartYearFocused, isEndYearFocused, isActive;
+
```
Comment (eivenlour) [inline lib/profile/save_profile/sections/education/education_item.dart:39]: Echoing my past comment to delete this line

File: lib/profile/save_profile/sections/education/education_item.dart [hunk 19]
```diff
@@ -67,16 +71,31 @@ class _SaveEducationItemState extends State<SaveEducationItem> {
...
+            onPressed: () {
+              handleClose();
+            },
+            icon: Icon(
+                Icons.arrow_back_ios,
+                color: Theme.of(context).accentColor
+              )
+            ),
+            backgroundColor: Colors.white,
+          ),
```
Comment (eivenlour) [inline lib/profile/save_profile/sections/education/education_item.dart:98]: Fix the indentation here please

File: lib/profile/save_profile/sections/education/education_item.dart [hunk 20]
```diff
@@ -331,19 +305,112 @@ class _SaveEducationItemState extends State<SaveEducationItem> {
...
+  Widget buildAddButton() {
+    return Container(
+      padding: EdgeInsets.only(top:100.0, bottom: 20.0, 
+      left: 60.0, right: 60.0),
+      child: PrimaryButton(
+        isLight: true,
+        text: widget.editMode ? 'Edit' : 'Add',
+        onPressed: () {
+          handleSaveEducation();  
+        },
```
Comment (eivenlour) [inline lib/profile/save_profile/sections/education/education_item.dart:411]: This could be changed to `onPressed: handleSaveEducation`

**Comment by eivenlour** [comment]
> Also, the fields, when focused, should have the inner icon in the teal colour as well!

Were you able to address this comment?

**Comment by xiangxinyue** [comment]
> > Also, the fields, when focused, should have the inner icon in the teal colour as well!
> 
> Were you able to address this comment?

I think I need explanation for that due to the limit understanding for this comment

**Pull request merged**